src/components/shared/inheritedComponents/ViewFormBase.tsx

```diff
@@ -9,12 +10,35 @@ interface IViewFormBaseState {
 class ViewFormBase<Props, State extends IViewFormBaseState> extends Component<Props, State> {
     protected isFormValid = () => {
         return !Object.values(this.state.invalidPropertiesMap)
-            .some(fieldIsValid => !fieldIsValid);
+            .some(validatorResult => !validatorResult.isValid);
     }
 
-    protected markInvalidProperties = (property: string, isValid: boolean) => {
+    protected validateAllProperties = (validationModel: object, validationEntity: any) => {
+        this.setState({
+            invalidPropertiesMap: {},
+        });
+
+        for (const property of Object.keys(validationModel)) {
```

Comment:
Object.entries(validationModel).forEach(([key, value]) => {
// do something
});

Reply:
Looping through entries now. ValidateProperty now takes validatorRule instead of validationModel.